host/src/clocks.rs

```diff
@@ -70,4 +70,17 @@ impl wasi_clocks::WasiClocks for WasiCtx {
             nanoseconds: res.subsec_nanos(),
         })
     }
+
+    async fn close_monotonic_clock(
+        &mut self,
+        clock: wasi_clocks::MonotonicClock,
+    ) -> anyhow::Result<()> {
+        self.table_mut().delete(clock);
```

Comment:
Should these perhaps verify that the value being removed is indeed of the expected type? (e.g. preventing this from being used to close a file)

Reply:
I've updated `Table::delete` to take a type parameter for verification -- the code here just needs to be updated to provide that parameter.